stdlib/args.ysh

```diff
@@ -167,5 +190,26 @@ func parseArgs(spec, argv) {
     setvar args[spec.rest] = rest
   }
 
+  # Set defaults for flags and args
+  for flag in (spec.flags) {
+    if (flag.name not in args) {
+      if (flag.required) {
```

Comment:
I'm not sure we should have the idea of "required flags"

I believe most command line programs follow the convention that

- "required" things are ALWAYS args
- but some args might not be required
- flags are always optional

If you look at `grep`, that's true

```
grep mypat  # stdin

grep mypat MYFILE

grep -C 1 mypat  # optional flag

```

i.e. there are no required flags

I think it's OK for the user to add their own validation with a BETTER error in case they want required flags, like

```
if (flag.message === null) {
  error 'Require --message' (status=2)
}
```